src/frontends/tensorflow/src/op/interpolate.cpp

```diff
@@ -13,34 +13,43 @@ namespace frontend {
 ov::OutputVector translate_interpolate_op(const NodeContext& node) {
-    auto input = node.get_input(0);
-    auto input_sizes = node.get_input(1);
+    default_op_checks(node, 2, {"ResizeBilinear", "ResizeNearestNeighbor"});
+    auto images = node.get_input(0);
+    auto size = node.get_input(1);
 
+    // retrieve optional attribute
+    auto tf_align_corners = node.get_attribute<bool>("align_corners", false);
+
+    // prepare attributes for OpenVINO Interpolate operation
     Interpolate::InterpolateAttrs interpolate_attrs;
     interpolate_attrs.mode = Interpolate::InterpolateMode::LINEAR;
     interpolate_attrs.shape_calculation_mode = Interpolate::ShapeCalcMode::SIZES;
-    if (node.get_attribute<bool>("align_corners", false))
+    if (tf_align_corners) {
         interpolate_attrs.coordinate_transformation_mode = Interpolate::CoordinateTransformMode::ALIGN_CORNERS;
+    }
 
     if (node.get_op_type() == "ResizeNearestNeighbor") {
         interpolate_attrs.mode = Interpolate::InterpolateMode::NEAREST;
         interpolate_attrs.nearest_mode = Interpolate::NearestMode::ROUND_PREFER_FLOOR;
     }
 
-    // TODO: do we need this .get_shape() actually?
-    auto input_shape = input.get_shape();
-    std::vector<float> spatial_shape = {static_cast<float>(input_shape[1]), static_cast<float>(input_shape[2])};
-    auto ng_spatial_shape = make_shared<Constant>(element::f32, Shape{2}, spatial_shape);
-
-    auto ng_sizes = make_shared<Convert>(input_sizes, element::f32);
-    auto ng_scales = make_shared<Divide>(ng_sizes, ng_spatial_shape);
-    auto ng_axes = make_shared<Constant>(element::i32, Shape{2}, std::vector<int>({2, 3}));
-
-    input = make_transpose(input, {0, 3, 1, 2});
-    auto res = make_shared<Interpolate>(input, input_sizes, ng_scales, ng_axes, interpolate_attrs)->output(0);
-    res = make_transpose(res, {0, 2, 3, 1});
-    set_node_name(node.get_name(), res.get_node_shared_ptr());
-    return {res};
+    // prepare scales input
+    auto images_shape = make_shared<ShapeOf>(images, ov::element::i32);
+    auto spatial_shape = make_shared<Slice>(images_shape,
+                                            make_shared<Constant>(element::i64, Shape{1}, std::vector<int64_t>{1}),
+                                            make_shared<Constant>(element::i64, Shape{1}, std::vector<int64_t>{3}),
+                                            make_shared<Constant>(element::i64, Shape{1}, std::vector<int64_t>{1}),
+                                            make_shared<Constant>(element::i64, Shape{1}, std::vector<int64_t>{0}));
+    auto scales = make_shared<Divide>(make_shared<Convert>(size, element::f32),
+                                      make_shared<Convert>(spatial_shape, element::f32));
+
+    // since Interpolate is layout agnostic
+    // we can avoid Transpose operation by specifying axes = {1, 2} for original NHWC layout
+    auto axes = make_shared<Constant>(element::i32, Shape{2}, std::vector<int>({1, 2}));
```

Comment:
Can this operation be in NCHW layout? Also I am not sure about tf1 ResizeBilinear and ResizeNearestNighbor, but resize from tf2 also support 3D inputs (no batch), not only 4D (https://www.tensorflow.org/api_docs/python/tf/image/resize), can we check that in case the rank is static it is 4D?

Reply:
This is Python API that can wrap or hide the whole sub-graph. It is better to look at C++ where one C++ object represents one operation in the graph.